src/solver/variable/economy/max-mrg-utils.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+** Copyright 2007-2023 RTE
+** Authors: Antares_Simulator Team
+**
+** This file is part of Antares_Simulator.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the GNU General Public License as published by
+** the Free Software Foundation, either version 3 of the License, or
+** (at your option) any later version.
+**
+** There are special exceptions to the terms and conditions of the
+** license as they are applied to this software. View the full text of
+** the exceptions in file COPYING.txt in the directory of this software
+** distribution
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** GNU General Public License for more details.
+**
+** You should have received a copy of the GNU General Public License
+** along with Antares_Simulator. If not, see <http://www.gnu.org/licenses/>.
+**
+** SPDX-License-Identifier: licenceRef-GPL3_WITH_RTE-Exceptions
+*/
+
+#include <yuni/yuni.h>
+#include <antares/study/study.h>
+#include "max-mrg.h"
+#include <antares/study/area/scratchpad.h>
+
+const unsigned int nbHoursInWeek = 168;
+
+namespace Antares::Solver::Variable::Economy
+{
+
+MaxMrgDataFactory::MaxMrgDataFactory(const State &state, unsigned int numSpace) :
+        state_(state),
+        numSpace_(numSpace),
+        weeklyResults_(state.problemeHebdo->ResultatsHoraires[state.area->index])
+{
+    maxMRGinput_.hydroGeneration = weeklyResults_.TurbinageHoraire.data();
+    maxMRGinput_.hydroMaxPower = state_.area->hydro.maxPower[Data::PartHydro::genMaxP];
+    maxMRGinput_.dtgMargin = state_.area->scratchpad[numSpace].dispatchableGenerationMargin;
+    maxMRGinput_.hourInYear = state.hourInTheYear;
+    maxMRGinput_.calendar = &state.study.calendar;
+    maxMRGinput_.areaName = state_.area->name.c_str();
+}
+
+MaxMRGinput MaxMrgUsualDataFactory::data()
+{
+    if (state_.simplexRunNeeded)
+    {
+        maxMRGinput_.spillage = weeklyResults_.ValeursHorairesDeDefaillanceNegative.data();
+    }
+    else
+    {
+        maxMRGinput_.spillage = state_.resSpilled[state_.area->index];
+    }
+
+    maxMRGinput_.dens = weeklyResults_.ValeursHorairesDeDefaillancePositive.data();
+    return maxMRGinput_;
+}
+
+MaxMRGinput MaxMrgCSRdataFactory::data()
+{
+    maxMRGinput_.spillage = weeklyResults_.ValeursHorairesDeDefaillanceNegative.data();
+    maxMRGinput_.dens = weeklyResults_.ValeursHorairesDeDefaillancePositiveCSR.data();
+    return maxMRGinput_;
+}
+
+void computeMaxMRG(double* opmrg, const MaxMRGinput& in)
```

Comment:
Note that we now call std algorithm instead of legacy code that did the same thing : 
-  `std::accumulate`
- `std::min_element`
- ` std::mx_element`

It makes the code shorter and clearer, but introduces very small rounding errors